snap/hooks/configure

```diff
@@ -52,6 +61,28 @@ handle_svc () {
     esac
 }
 
+restart_db_services () {
+    # since the configuration files have been modified, those
+    # changes need to be progpogated into relevant services, so
+    # restart the services with updated configuration if they were
+    # already running
+    for svc in ${DB_SERVICES[*]}; do
+        status=`snapctl services "$SNAP_NAME.$svc" | grep -o "active"`
+
+        if [ "$status" = "active" ]; then
+            snapctl restart "$SNAP_NAME.$svc"
+        fi
+    done
+}
+
+update_consul_db_type () {
+    curl -s --request PUT --data "$2" \
```

Comment:
Why Consul and not env overrides? 

Remember that env overrides are record of truth now, not consul.

Reply:
Well... back when core-config-seed still was around, this code modified the configuration.toml files in **$SNAP_DATA**, stopped the affected services, ran core-config-seed w/the overwrite flag, and then restarted the services. Doing this on a per-service basis seemed a bit too complex. Yes, it would be possible to do this via env overrides, but that would require an additional wrapper script (see bin/security-secret-store-env-var.sh which is used to handle vault being enabled/disabled). When Ian and I first discussed how to do this, he suggested just modifying the values directly in Consul, so I just went with that approach.

My gut tells me that modifying the dbtype (which needs to be done *before* anything's been written to the db) is going to be a fairly uncommon operation. The only reason I've kept it around is that we still haven't officially dropped mongo yet.

If you feel strongly about this, I'll go ahead and create the wrapper scripts.

Reply:
Do you have a mode where Consul isn't used? If not, then I am fine with this approach.

Reply:
Since we don't yet support running without consul, I'd say leave it like this. Setting env vars for other services is also awkward because you need to pass state to those other services from here via either snap config or a file or something so it's not quite as simple as just setting the env var.